pkg/cli/apk.go

```diff
@@ -100,3 +120,174 @@ func Index() *cobra.Command {
 	cmd.Flags().StringVar(&repo, "repo", "wolfi", "repo to get packages from")
 	return cmd
 }
+
+var buckets = map[string]string{
+	"wolfi":  "gs://wolfi-production-registry-destination/os",
+	"stage1": "gs://wolfi-production-registry-destination/bootstrap/stage1",
+	"stage2": "gs://wolfi-production-registry-destination/bootstrap/stage2",
+	"stage3": "gs://wolfi-production-registry-destination/bootstrap/stage3",
+}
+
+func GenerateIndex() *cobra.Command {
+	var arch, bucket, signingKey string
+	var publish bool
+	cmd := &cobra.Command{
+		Use: "generate-index",
+		Long: `This command generates an APKINDEX from the contents of a remote bucket.
+
+Specify the bucket with --bucket. The default is "wolfi", the main prod Wolfi bucket.
+Other acceptable values include "stage1", "stage2" and "stage3" for the bootstrap buckets.
+Otherwise, specify any GCS bucket location with the gs:// prefix.
+
+If --signing-key is passed, the APKINDEX will be signed with that key.
+
+If --publish is passed, the APKINDEX will be published back to the bucket.
+Otherwise it's written to APKINDEX.tar.gz.
+`,
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			// Map a friendly string like "wolfi" to its bucket.
+			if got, found := buckets[bucket]; found {
+				bucket = got
+			}
+			if !strings.HasPrefix(bucket, "gs://") {
+				return errors.New("--bucket must have gs:// prefix")
+			}
+
+			if signingKey != "" {
+				k, err := os.Open(signingKey)
+				if err != nil {
+					return err
+				}
+				defer k.Close()
+			}
+
+			idx := &repository.ApkIndex{
+				// TODO: sign the index.
+				// https://github.com/chainguard-dev/melange/blob/main/internal/sign/sign.go#L34
```

Comment:
is it not done below?

Reply:
It's true! Removed the TODO.

I also removed the description here, since it wasn't adding much. Our current index is generated without a description.